packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkImplementation.kt

```diff
@@ -39,7 +39,23 @@ class DdSdkImplementation(
 
         val nativeInitialization = DdSdkNativeInitialization(appContext, datadog)
         nativeInitialization.initialize(ddSdkConfiguration)
-        monitorJsRefreshRate(ddSdkConfiguration)
+
+        this.frameRateProvider = createFrameRateProvider(ddSdkConfiguration)
+
+        reactContext.addLifecycleEventListener(object : LifecycleEventListener {
+            override fun onHostResume() {
+                frameRateProvider?.start()
+            }
+
+            override fun onHostPause() {
+                frameRateProvider?.stop()
+            }
+
+            override fun onHostDestroy() {
+                frameRateProvider?.stop()
+            }
```

Comment:
are these callbacks always called on the main thread?

Reply:
Yes, they are. For reference: https://github.com/facebook/react-native/blob/7fd6c2c45e532417d7ebb3fbeba3360cb6905ab3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/ReactContext.java#L211-L224